src/code/z_actor.c

```diff
@@ -457,439 +1244,3461 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_InsertTranslation(0.0f, entry->yShift, 500.0f, MTXMODE_APPLY);
+    Matrix_CopyCurrentState(&baseMtxF);
+
+    chainsTranslateX = __sinf(entry->chainAngle - chainRotZ) * -(10 - frame) * 0.1f * entry->chainLength;
+    chainsTranslateY = __cosf(entry->chainAngle - chainRotZ) * (10 - frame) * 0.1f * entry->chainLength;
+
+    for (i = 0; i < 4; i++) {
+        Matrix_SetCurrentState(&baseMtxF);
+        Matrix_InsertZRotation_f(chainRotZ, MTXMODE_APPLY);
+        Matrix_InsertTranslation(chainsTranslateX, chainsTranslateY, 0.0f, MTXMODE_APPLY);
+        if (entry->chainsScale != 1.0f) {
+            Matrix_Scale(entry->chainsScale, entry->chainsScale, entry->chainsScale, MTXMODE_APPLY);
+        }
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, entry->chainDL);
+
+        if ((i % 2) != 0) {
+            rotZStep = 2.0f * entry->chainAngle;
+        } else {
+            rotZStep = M_PI - (2.0f * entry->chainAngle);
+        }
+
+        chainRotZ += rotZStep;
+    }
+
+    Matrix_SetCurrentState(&baseMtxF);
+    Matrix_Scale(frame * 0.1f, frame * 0.1f, frame * 0.1f, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, entry->lockDL);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void Actor_SpawnShieldParticlesMetal(GlobalContext* globalCtx, Vec3f* pos) {
+    CollisionCheck_SpawnShieldParticlesMetal(globalCtx, pos);
+}
+
+void Actor_SetColorFilter(Actor* actor, u16 colorFlag, u16 colorIntensityMax, u16 xluFlag, u16 duration) {
+    if ((colorFlag == 0x8000) && !(colorIntensityMax & 0x8000)) {
+        Actor_PlaySfxAtPos(actor, NA_SE_EN_LIGHT_ARROW_HIT);
+    }
+
+    actor->colorFilterParams = colorFlag | xluFlag | ((colorIntensityMax & 0xF8) << 5) | duration;
+    actor->colorFilterTimer = duration;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE680.s")
+Hilite* func_800BCBF4(Vec3f* arg0, GlobalContext* globalCtx) {
+    Vec3f lightDir;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BF7CC.s")
+    lightDir.x = globalCtx->envCtx.dirLight1.params.dir.x;
+    lightDir.y = globalCtx->envCtx.dirLight1.params.dir.y;
+    lightDir.z = globalCtx->envCtx.dirLight1.params.dir.z;
+
+    return func_800B7FE0(arg0, &globalCtx->view.eye, &lightDir, globalCtx->state.gfxCtx);
+}
+
+Hilite* func_800BCC68(Vec3f* arg0, GlobalContext* globalCtx) {
+    Vec3f lightDir;
+
+    lightDir.x = globalCtx->envCtx.dirLight1.params.dir.x;
+    lightDir.y = globalCtx->envCtx.dirLight1.params.dir.y;
+    lightDir.z = globalCtx->envCtx.dirLight1.params.dir.z;
+
+    return func_800B8018(arg0, &globalCtx->view.eye, &lightDir, globalCtx->state.gfxCtx);
+}
+
+void func_800BCCDC(Vec3s* points, s32 pathCount, Vec3f* pos1, Vec3f* pos2, s32 arg4) {
```

Comment:
This seems to be some kind of variation on the function `EnRiverSound_GetSoundPos` that I've recently looked at from OoT:
https://github.com/zeldaret/oot/blob/2e146e865aa0c67bfa99dae3ee534f67e7a599ec/src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c#L115

MM's river sounds seems to have moved it's comparable river path calculations to this function instead of `EnRiverSound_GetSoundPos` and `EnRiverSound_FindClosestPointOnLineSegment` from OoT. 

I'll document this function and submit a PR to either Anghelo or Master depending on whenever this gets in. This shouldn't hold up this PR though